### PR TITLE
Counter pr on https://github.com/mdn/yari/pull/1682

### PR DESCRIPTION
**Pull request by peterbe**
@diminutivesloop  What do you think of this?
It seems to work. I tested it with the real content as well as the `testing/content/files` and it seems to behave nicely. 
But it was a bit of a rushed thing so perhaps you can look it over.

### CODE REVIEW

**Comment by peterbe** [comment]
What I would expect is that you look over this PR and test it and check for weird corner cases we haven't considered. 
And if you're happy, merge it into your branch (regular merge) and we can refocus our attention on https://github.com/mdn/yari/pull/1682 for a final review. 
Yes, we'll merge it and my contribution disappears but I don't care. You'll get the commit-credit and you deserve it.

**Comment by diminutivesloop** [comment]
So I tested searches for a few pages on MDN and I didn't notice any significant regression in results quality for either this PR or my original PR. I'm happy to merge your changes into my PR as you suggested although I am curious what motivated the changes since they seem to make the code more complicated and brittle without changing any functionality.

**Comment by peterbe** [comment]
> more complicated and brittle without changing any functionality.

Excellent feedback. Clearly your actually reading the code changes. 
Well, I would argue that the functionality changed. Now the `/` isn't highlighted anymore which makes sense because it was the "activation character" rather than something you're actually searching for. 
So in that sense, it solves the original issue. 

But if you think the whole thing is getting overly complicated and brittle perhaps the best thing is to go back to the drawing board and start afresh on a different approach. 
Eager to hear your thoughts.

**Comment by diminutivesloop** [comment]
Well my question wasn't whether either PR introduces new functionality but rather why you think these changes are a better fix than my original PR. I think the additional comments in `getSubstringsFromRanges` are helpful but the other changes replace a simple `inputValue.substring(1)` with three calls to `slice()`. Sorry if I'm bikeshedding a farily unimportant piece of code and again I'm happy to merge your changes into my PR if you still think they're worth it.

**Comment by peterbe** [comment]
The reason I did this PR was because your PR wasn't working correctly. You had to type `/wbfoo` for it find enough matches. That's what started it. Now, I'm starting to understand why. The code that puts a score on the match stays at 0 if there are too few matches. When you remove the `/` from the search, it was actually searching for `wboo` and apparently wasn't enough. So, I started venturing on keeping the `/` for the matching and then did the necessary "tricks" so it would highlight independent of the `/` prefix. 

But it makes me wonder; is it a mistake to actually do the fuzzy search based on `/otherchars` or would it be better to use `otherchars` only. 

Perhaps the more interesting question is; what the heck is wrong with `wboo`?? Why doesn't that think the `en-us/docs/WeB/fOO` is a good enough match?!

> Sorry if I'm bikeshedding a farily unimportant piece of code

Don't be sorry. We're learning to work together.

File: client/src/fuzzy-search.ts [hunk 1]
```diff
@@ -60,7 +60,11 @@ export class FuzzySearch {
     // each match's ranges so we can finally result a list of sub-strings.
     const res: CombinedMatchResult[] = candidates
       .map((doc) => {
-        const matched = match(needle, doc.url, {
+        // The needle will always start with a `/` and so will always the `doc.url`
+        // but it's not interesting to match on it. So, essentially,
+        // match `wboo` with `en-US/docs/Web/Foo`
+        // instead of `/wboo` with `/en-US/docs/Web/Foo`
+        const matched = match(needle.slice(1), doc.url.slice(1), {
```
Comment (diminutivesloop) [inline client/src/fuzzy-search.ts:67]: Ah so I was confused because you didn't revert my changes to `search.test.js` so I didn't realize the old test now passes with your changes. Looks like the key difference is here where you remove the leading slash from the url while I only remove it from the query.
